pkg/cache/v2/simple.go

```diff
@@ -94,11 +100,56 @@ type snapshotCache struct {
 // Logger is optional.
 func NewSnapshotCache(ads bool, hash NodeHash, logger log.Logger) SnapshotCache {
 	return &snapshotCache{
-		log:       logger,
-		ads:       ads,
-		snapshots: make(map[string]Snapshot),
-		status:    make(map[string]*statusInfo),
-		hash:      hash,
+		log:               logger,
+		ads:               ads,
+		snapshots:         make(map[string]Snapshot),
+		status:            make(map[string]*statusInfo),
+		hash:              hash,
+		heartbeatRoutines: make(map[string]heartbeatHandle),
+	}
+}
+
+func (cache *snapshotCache) sendHeartbeats(ctx context.Context, wg *sync.WaitGroup, node string) {
+	snapshot := cache.snapshots[node]
+	if info, ok := cache.status[node]; ok {
+		info.mu.Lock()
+		for id, watch := range info.watches {
```

Comment:
Can this be done concurrently? I wonder if we'll run into performance problems doing this regularly for all watches.

Reply:
Hmm did you have any specific strategy in mind? We would still need to grab the lock to read/delete the watches so I'm not sure where we'd benefit from the concurrency here

Thinking about this some more, one improvement (somewhat in line with the TODO) would be to track the watches per type, which would allow us to look at which types have TTL'd resources and only iterate over the associated watches. A further optimization would be to track whether resources have seen an update recently and skip the heartbeat if so, which would reduce heartbeating during times of high cache activity. Not sure how much of this we'd want in an intial PR, so lmk what you think